src/tests/suite/dfs_sys_unit_test.c

```diff
@@ -0,0 +1,802 @@
+
+	/** Get xattr on sym1->file1 */
+	rc = dfs_sys_getxattr(dfs_sys_mt, sym1, name1, buf, &buf_size, 0);
+	assert_int_equal(rc, 0);
+	assert_int_equal(buf_size, size);
+	assert_string_equal(buf, val1);
+
+	/** Get xattr on sym1 itself */
+	rc = dfs_sys_getxattr(dfs_sys_mt, sym1, name2, buf, &buf_size,
+			      O_NOFOLLOW);
+	assert_int_equal(rc, 0);
+	assert_int_equal(buf_size, size);
+	assert_string_equal(buf, val2);
+
+	/** Remove xattr on sym1->file1 */
+	rc = dfs_sys_removexattr(dfs_sys_mt, sym1, name1, 0);
+	assert_int_equal(rc, 0);
+	rc = dfs_sys_listxattr(dfs_sys_mt, sym1, buf, &buf_size, 0);
+	assert_int_equal(rc, 0);
+	assert_int_equal(buf_size, 0);
+
+	/** Remove xattr on sym1 itself */
+	rc = dfs_sys_removexattr(dfs_sys_mt, sym1, name2, O_NOFOLLOW);
+	assert_int_equal(rc, 0);
+	rc = dfs_sys_listxattr(dfs_sys_mt, sym1, buf, &buf_size, O_NOFOLLOW);
+	assert_int_equal(rc, 0);
+	assert_int_equal(buf_size, 0);
+
+	delete_simple_tree(dir1, file1, sym1);
+}
+
+static const struct CMUnitTest dfs_sys_unit_tests[] = {
+	{ "DFS_SYS_UNIT_TEST1:  DFS Sys mount / umount",
+	  dfs_sys_test_mount, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST2:  DFS Sys get_dfs_obj",
+	  dfs_sys_test_get_dfs, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST3:  DFS Sys create / remove",
+	  dfs_sys_test_create_remove, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST4:  DFS Sys access / chmod",
+	  dfs_sys_test_access_chmod, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST5:  DFS Sys open / stat",
+	  dfs_sys_test_open_stat, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST6:  DFS Sys readlink",
+	  dfs_sys_test_readlink, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST7:  DFS Sys setattr",
+	  dfs_sys_test_setattr, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST8:  DFS Sys read / write",
+	  dfs_sys_test_read_write, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST9:  DFS Sys opendir / readdir",
+	  dfs_sys_test_open_readdir, async_disable, test_case_teardown},
+	{ "DFS_SYS_UNIT_TEST10: DFS Sys xattr",
+	  dfs_sys_test_xattr, async_disable, test_case_teardown},
+};
+
+static int
+dfs_sys_setup(void **state)
+{
+	test_arg_t		*arg;
+	int			rc = 0;
+
+	rc = test_setup(state, SETUP_POOL_CONNECT, true, DEFAULT_POOL_SIZE,
+			0, NULL);
+	assert_int_equal(rc, 0);
+
+	arg = *state;
+
+	if (arg->myrank == 0) {
+		uuid_generate(co_uuid);
+		rc = dfs_cont_create(arg->pool.poh, co_uuid, NULL, &co_hdl,
+				     NULL);
+		assert_int_equal(rc, 0);
+		printf("Created DFS Container "DF_UUIDF"\n", DP_UUID(co_uuid));
+		rc = dfs_sys_mount(arg->pool.poh, co_hdl, O_RDWR, 0,
+				   &dfs_sys_mt);
+		assert_int_equal(rc, 0);
+	}
+
+	handle_share(&co_hdl, HANDLE_CO, arg->myrank, arg->pool.poh, 0);
+	dfs_sys_test_share(arg->pool.poh, co_hdl, arg->myrank, 0, &dfs_sys_mt);
+
+	return rc;
+}
+
+static int
+dfs_sys_teardown(void **state)
+{
+	test_arg_t	*arg = *state;
+	int		rc;
+
+	rc = dfs_sys_umount(dfs_sys_mt);
```

Comment:
This implicitly makes sure the hash is cleaned up, but in the future we should probably add hash-specific tests to make sure it is behaving as expected.